### PR TITLE
Introduce sdk errors

### PR DESCRIPTION
**Pull request by xitronix**

### CODE REVIEW

File: universal-login-sdk/lib/observers/BalanceObserver.ts [hunk 1]
```diff
@@ -13,7 +14,7 @@ export class BalanceObserver extends ObserverRunner {
   }
 
   async startAndSubscribe(contractAddress: string, callback: BalanceChangedCallback) {
-    ensure(!this.isRunning(), Error, 'Other wallet waiting for counterfactual deployment. Stop BalanceObserver to cancel old wallet instantialisation.');
+    ensure(!this.isRunning(), BalanceObserverConfilct);
```
Comment (marekkirejczyk) [inline universal-login-sdk/lib/observers/BalanceObserver.ts]: BalanceObserverConf**li**ct

File: universal-login-sdk/lib/utils/errors.ts [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+    super(message, errorType);
+    this.errorType = errorType;
+    Object.setPrototypeOf(this, ValidationFailed.prototype);
+  }
+}
+
+export class InvalidBytecode extends ValidationFailed {
+  constructor () {
+    super('Proxy Bytecode is not supported by relayer', 'InvalidBytecode');
+    Object.setPrototypeOf(this, BalanceObserverConfilct.prototype);
```
Comment (marekkirejczyk) [inline universal-login-sdk/lib/utils/errors.ts]: Object.setPrototypeOf(this, **InvalidBytecode**.prototype);

File: universal-login-sdk/lib/utils/errors.ts [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+  constructor () {
+    super('Transaction hash is not found in Message Status', 'TransactionHashNotFound');
+    Object.setPrototypeOf(this, TransactionHashNotFound.prototype);
+  }
+}
+
+export class TimeoutError extends SDKError {
+  constructor () {
+    super('Timeout exceeded', 'TransactionHashNotFound');
+    Object.setPrototypeOf(this, TransactionHashNotFound.prototype);
```
Comment (marekkirejczyk) [inline universal-login-sdk/lib/utils/errors.ts]: Object.setPrototypeOf(this, **TimeoutError**.prototype);

File: universal-login-sdk/lib/sdk.ts [hunk 4]
```diff
@@ -153,12 +154,12 @@ class UniversalLoginSDK {
   }
 
   async getBalanceObserver() {
-    ensureNotNull(this.config, Error, 'Relayer configuration not yet loaded');
+    ensureNotNull(this.config, RelayerConfigNotFound);
```
Comment (marekkirejczyk) [inline universal-login-sdk/lib/sdk.ts]: `RelayerConfigNotFound` -> 
`MissingConfiguration`

File: universal-login-sdk/lib/observers/DeploymentObserver.ts [hunk 5]
```diff
@@ -32,7 +33,7 @@ export class DeploymentObserver extends ObserverRunner {
   private async checkContract(futureContractAddress: string){
     const bytecode = await this.blockchainService.getCode(futureContractAddress);
     if (isContractExist(bytecode)){
-      ensure(this.contractWhiteList.proxy.includes(utils.keccak256(bytecode)), Error, 'Proxy Bytecode is not supported by relayer');
+      ensure(this.contractWhiteList.proxy.includes(utils.keccak256(bytecode)), InvalidBytecode);
```
Comment (marekkirejczyk) [inline universal-login-sdk/lib/observers/DeploymentObserver.ts]: `InvalidBytecode` -> `UnsupportedBytecode`

File: universal-login-sdk/lib/observers/DeploymentObserver.ts [hunk 6]
```diff
@@ -16,7 +17,7 @@ export class DeploymentObserver extends ObserverRunner {
   }
 
   startAndSubscribe(futureContractAddress: string, callback: OnContractDeployed) {
-    ensure(!this.isRunning(), Error, 'DeploymentObserver is waiting for contract deployment. Stop observer to cancel waiting');
+    ensure(!this.isRunning(), DeploymentObserverConfilct);
```
Comment (marekkirejczyk) [inline universal-login-sdk/lib/observers/DeploymentObserver.ts]: `DeploymentObserverConfilct` -> `ConcurrentDeployment`

Comment (xitronix) [reply universal-login-sdk/lib/observers/DeploymentObserver.ts]: DeploymentObserverConfilct and BalanceObserverConflict merge into one type error?

Comment (marekkirejczyk) [reply universal-login-sdk/lib/observers/DeploymentObserver.ts]: Sounds good, same reason, same error.

**Comment by marekkirejczyk** [comment]
Rebase?

**Pull request merged**